frb_dart/lib/src/ffi.dart

```diff
@@ -62,7 +66,11 @@ abstract class FrbOpaque extends FrbOpaqueBase {
   @internal
   PlatformPointer share() {
     if (!isStale()) {
-      return shareFn(_ptr);
+      var ptr = shareFn(_ptr);
+      if (_move) {
+        dispose();
+      }
+      return ptr;
```

Comment:
Hmm the name (or the API) seems to be improveable: We say we are "share"ing something, but we are indeed "moving".

Reply:
shareOrMove looks OK to me (though still a bit weird :/)

getter seems not to satisfy flutter/dart style, since a getter may not be this heavy and creates a new reference count etc